kratos/includes/parallel_environment.h

```
@@ -114,8 +115,31 @@ class KRATOS_API(KRATOS_CORE) ParallelEnvironment
      * @param rModelPart Model part to which the fill communicator will be applied
      * @return FillCommunicator::Pointer Pointer to the new fill communicator instance
      */
+    KRATOS_DEPRECATED_MESSAGE("This function is deprecated, please use \"CreateFillCommunicatorFromGlobalParallelism\" instead")
     static FillCommunicator::Pointer CreateFillCommunicator(ModelPart& rModelPart);
 
+    /**
+     * @brief Create a fill communicator object
+     * This method uses the previously registered fill communicator factory for the creation of a new fill communicator pointer
+     * @param rModelPart Model part to which the fill communicator will be applied
+     * @param rDataCommunicatorName Name of the data communicator to be retrieved for the fill communicator construction
+     * @return FillCommunicator::Pointer Pointer to the new fill communicator instance
+     */
+    static FillCommunicator::Pointer CreateFillCommunicatorFromGlobalParallelism(
```

Just asking, why not overloading the already existent `CreateFillCommunicator`?

---

as before the point is not to silently pass around the default data-comm

---

OK. Makes sense.

---

Should we remove the old signature usages in this PR? (there should be only a few)

---

I deprecated them so that it can be done safely in separate PRs
I think you are the only user so far, but maybe someone else uses them already in a branch
Esp when updating the Core I fell safer doing it the soft way to not break ppls code